### PR TITLE
Replace usages of LogEvent default constructor

### PR DESCRIPTION
**Pull request by tenevdev**
#### Description
The default constructor doesn't initialise the Data field because this is a struct. This may cause NPEs. Calling the non-default constructor is a temporary solution.

### CODE REVIEW

**Comment by tenevdev** [comment]
As stated in the description - this is a temporary solution. I'd like to raise another PR which makes it impossible to make this mistake again. However, this will require structural changes to how we log events. The performance gained having log event be a struct is unclear. Let me know if anyone has data supporting this design decision.

**Comment by jamiebrynes7** [comment]
We are planning a refactor to logging soon ™️ so we can add this onto the list of things to revisit

File: workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs [hunk 1]
```diff
@@ -271,7 +271,7 @@ private void RunWithExceptionHandling(Action action)
             catch (Exception e)
             {
                 // Log the exception but do not rethrow it.
-                logger.HandleLog(LogType.Exception, new LogEvent().WithException(e));
+                logger.HandleLog(LogType.Exception, new LogEvent("Caught exception").WithException(e));
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs]: `Caught exception while executing an action` 

If we know these are only used in callbacks in the Monobehaviour workflow we could even be more explicit?

Comment (tenevdev) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs]: I just didn't want to make any assumptions about what the caller is doing. Also "while executing action" seemed redundant, the user will see a stack trace anyway.

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs]: The content of the action could throw, but these actions are all callbacks registered against a Reader/Writer right now I think

File: workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs [hunk 2]
```diff
@@ -271,7 +271,7 @@ private void RunWithExceptionHandling(Action action)
             catch (Exception e)
             {
                 // Log the exception but do not rethrow it.
-                logger.HandleLog(LogType.Exception, new LogEvent("Caught exception").WithException(e));
+                logger.HandleLog(LogType.Exception, new LogEvent("Caught exception in MonoBehaviour").WithException(e));
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs]: `in a MonoBehaviour`

File: workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReadersWriters/GameObjectDelegates.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public static class GameObjectDelegates
                 catch (Exception e)
                 {
                     // Log the exception but do not rethrow it, as other delegates should still get called
-                    logDispatcher.HandleLog(LogType.Exception, new LogEvent().WithException(e));
+                    logDispatcher.HandleLog(LogType.Exception, new LogEvent("Caught exception in a dispatcher callback").WithException(e));
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReadersWriters/GameObjectDelegates.cs]: dispatcher callback doesn't tell anything to a user

**Pull request merged**